weld/ast.rs

```diff
@@ -523,21 +522,22 @@ impl<T: TypeBounds> Expr<T> {
 
     /// Recursively transforms an expression in place by running a function on it and optionally replacing it with another expression.
     pub fn transform_and_continue<F>(&mut self, func: &mut F)
-        where F: FnMut(&mut Expr<T>) -> Option<(Expr<T>, bool)>
+        where F: FnMut(&mut Expr<T>) -> (Option<Expr<T>>, bool)
     {
         match func(self) {
-            Some((e, true)) => {
+            (Some(e), true) => {
                 *self = e;
                 return self.transform_and_continue(func);
             }
-            Some((e, false)) => {
+            (Some(e), false) => {
                 *self = e;
             }
-            None => {
+            (None, true) => {
                 for c in self.children_mut() {
                     c.transform_and_continue(func);
                 }
             }
+            (None, false) => {}
```

Comment:
Hm, the reason for this is that if you do end up replacing self, you want to call the transform on yourself (to match on the replaced expression); otherwise, you call the transform on each child.